packages/typespec-ts/src/utils/clientUtils.ts

```diff
@@ -0,0 +1,18 @@
+import {
+  SdkClient,
+  SdkContext
+} from "@azure-tools/typespec-client-generator-core";
+import { listServices } from "@typespec/compiler";
+
+export function getRLCClients(dpgContext: SdkContext): SdkClient[] {
+  const services = listServices(dpgContext.program);
+
+  return services.map(service => {
+    return {
+      kind: "SdkClient",
+      name: service.type.name + "Client",
+      service: service.type,
+      type: service.type
+    }
+  });
```

Comment:
I found what Brian means about respect the package boundary is not to the respect the boundary that are defined by the @client decorator, but are the boundaries that this client.tsp refers to. so using `listServices` directly is the right choice here.